src/components/SearchBox/SearchBox.tsx

```diff
@@ -91,6 +91,13 @@ const SearchBox = React.forwardRef<HTMLInputElement, Props>(
       onSearch && onSearch();
     };
 
+    const onKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === "Enter" && internalRef.current.checkValidity()) {
```

Comment:
This should stop blur when the input contains an invalid value (https://github.com/canonical/react-components/pull/1065#issuecomment-2042285408)